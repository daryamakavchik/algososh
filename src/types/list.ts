import { ElementStates } from "./element-states";

export interface ILinkedList<T> {
  append: (element: T) => void;
  prepend: (element: T) => void;
  deleteHead: () => void;
  deleteTail: () => void;
  addByIndex: (element: T, index: number) => void;
  removeByIndex: (index: number) => void;
}

export type TListItem = {
  value: string;
  color: ElementStates;
};