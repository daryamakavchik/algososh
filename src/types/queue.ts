import { ElementStates } from "./element-states";
export interface IQueue<T> {
    enqueue: (item: T) => void;
    dequeue: () => void;
    peek: () => T | null;
  }

export type TQueueItem = {
    value?: string;
    color: ElementStates;
    head?: string;
  };