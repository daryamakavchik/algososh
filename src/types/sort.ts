import { ElementStates } from "./element-states";
export type TArray = {
    value: number;
    color: ElementStates;
  };