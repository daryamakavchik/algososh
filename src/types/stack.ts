import { ElementStates } from "./element-states";

export type TStackItem = {
  value: string;
  color: ElementStates;
};

export interface IStack<T> {
  push: (item: T) => void;
  pop: () => void;
  peek: () => T | null;
  getSize: () => number;
}
