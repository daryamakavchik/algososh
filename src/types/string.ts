import { ElementStates } from "./element-states";
export type TArray = {
    value: string;
    color: ElementStates;
  };
  