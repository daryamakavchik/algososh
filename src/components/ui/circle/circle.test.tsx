import React from "react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";
import * as renderer from "react-test-renderer";

describe("Круг рендерится без ошибок", () => {
  it("Круг без текста", () => {
    const tree = renderer.create(<Circle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг с буквами", () => {
    const tree = renderer.create(<Circle letter="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг с head", () => {
    const tree = renderer.create(<Circle head="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг с React-элементом в head", () => {
    const tree = renderer.create(<Circle head={<Circle />} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг с React-элементом в tail", () => {
    const tree = renderer.create(<Circle tail={<Circle />} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг с tail", () => {
    const tree = renderer.create(<Circle tail="tail" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг с index", () => {
    const tree = renderer.create(<Circle index={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг isSmall", () => {
    const tree = renderer.create(<Circle isSmall />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг default", () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг changing", () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Круг modified", () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
