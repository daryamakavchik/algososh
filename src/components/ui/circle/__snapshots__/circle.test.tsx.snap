// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Круг рендерится без ошибок Круг changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   changing"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Круг рендерится без ошибок Круг default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Круг рендерится без ошибок Круг isSmall 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle  small default"
    data-testid="topcircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Круг рендерится без ошибок Круг modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   modified"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Круг рендерится без ошибок Круг без текста 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Круг рендерится без ошибок Круг с React-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head"
      />
      <div
        className="circle   default"
        data-testid="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Круг рендерится без ошибок Круг с React-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head"
      />
      <div
        className="circle   default"
        data-testid="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>
</div>
`;

exports[`Круг рендерится без ошибок Круг с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="head"
  >
    test
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Круг рендерится без ошибок Круг с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="tail"
  />
</div>
`;

exports[`Круг рендерится без ошибок Круг с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="tail"
  >
    tail
  </div>
</div>
`;

exports[`Круг рендерится без ошибок Круг с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    >
      test
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;
